contracts/state/State.sol

```diff
@@ -118,6 +118,8 @@ contract State is Ownable2StepUpgradeable, IState {
         uint256[2][2] memory b,
         uint256[2] memory c
     ) public {
+        bytes2 idType = GenesisUtils.getIdType(id);
+        require(_stateData.isIdTypeSupported[idType], "id type is not supported");
```

Comment:
Repeatable code. Better to put it into some modifier or  public method.
It's interesting BTW whether the `isIdTypeSupported` method will be used by clients with `bytes2 idType` parameter in reality. Maybe clients in most cases would extract type id from identifier. If that is the case `uint256 id` can be used instead. What do you think @vmidyllic ?

Reply:
default type is fine, because onchain issuer during creation can't pass identifier but should check that id type is supported

Reply:
Agree on modifier.

Reply:
idType is used further in code, but I've added modifier also